### ISSUE TITLE
Metadata doesn't work and it's enable on the configuration file..

### ISSUE THREAD

**Issue by Playsky** [mikebrady/shairport-sync#345]
Hello !
I downloaded shairport-sync with metadata, I modified the configuration file and I enable the metadata with "yes" but when I launch shairport with "shairport-sync" in my command Line, and when I put my music, metadata doesn't appear in /tmp !!!

I just modified the .conf to put metadata to yes so I don't know why it doesn'work..

If you can help me that's will be wonderful !!
Thanks ans sorry for my English :)

**Comment by mikebrady** [comment]
Thanks for the post. This sometimes happens if `shairport-sync` does not have permission to write to `/tmp`. Could that be the problem?

**Comment by Playsky** [comment]
I saw the permissions of "tmp" ans it's wrote "every body" so I think it's ok, and I changed the directory with à folder which I created and I have the same problem..
Is there a thing to do to put permission to shairport-sync ? Sorry I'm not really strong with a raspberry :)

**Comment by mikebrady** [comment]
Is there anything in the log?

**Comment by Playsky** [comment]
No, when I execute the command "Shairport-sync" the only thing it's wrote in the command Line is:
"Successful Startup"

**Comment by mikebrady** [comment]
Thanks. If you set `log_verbosity` to 1 it will give some more information. 

Another thing is, if `/tmp/shairport-sync-metadata` already exists as a file, the pipe will not be created.

**Comment by Playsky** [comment]
Sorry for my late response !
So I modified log_verbosity but nothing Has changed !!
To be sure, the configuration file is shairport-sync/scripts/shairport-sync.conf ?
Because all changes I did didn't work and the name of the AirPlay what I put is the same of the name of my raspberry, so may be I don't modify the good file ! :)

**Comment by mikebrady** [comment]
That's very interesting – it makes me think that you have not completely installed `shairport-sync`. If you follow all the steps in the guide in [README.md](https://github.com/mikebrady/shairport-sync/blob/master/README.md), you will find a configuration file at `/etc/shairport-sync.conf`, and that is the file you must modify.

**Comment by Playsky** [comment]
Okay, so I re-installed shairport-sync Completly ;) so I modified the shairport.conf in /etc but again, nothing has changed, even the log_verbosity,  just  the name is good, which is the same as my raspberry..
And Shairport-sync lauch automatically so I can't launch shairport with the command line like before but it's normal !

**Comment by roblan** [comment]
Name is by default hostname now, so it looks like you are not modifying a right file.
You wrote shairport.conf in /etc when it should be shairport-sync.conf - is it a typo, or you've edited a wrong file?

**Comment by Playsky** [comment]
I'm sorry it's a typo... So in my Raspberry I have 3 shairport-sync.conf, one in /etc, an other in /etc but it's shairport-sync.conf.sample, and one in user/shairport-sync/scripts.
So in all of these files, I put the name "Test", to change with the name of my raspberry, and I put metadata enable to "yes". So I reboot my Raspberry but again I have the name of my raspberry and again, No metadata... So all of these files are wrong..

**Comment by Playsky** [comment]
Okay so I solve my problem ! I'm so silly !! I forgot to uncomment the lines in the configuration files !

But just I have a very important question. My really goal is to control airplay, so to pause the music, to start the next music etc.. I read a long time this page: http://nto.github.io/AirPlay.html#audio-remotecontrol but I still don't understand how does it work.. So I have: 
- The DACP-ID
- The Active-Remote
- User agent
But I don't know how to send this ?? I have to make a request but to which IP ? which port ? the 5000 ? And I try to send the three information sin a header in python but it doesn't work so how can I send these informations ? 
If you can give me an exemple, the url, or a python script exemple, That's will be PERFECT !

**Comment by joerg-krause** [comment]
For remote control to work you need to fetch the IP and port from the *_dacp._tcp* bonjour service. You get `DACP-ID` and `Active-Remote` from the metadata. Putting this together, you can build a HTTP request. Have a look at this example for one way to implement the DACP client: https://www.sugrsugr.com/index.php/airplay-prev-next/.

**Comment by mikebrady** [comment]
So, can we close this issue? It looks solved to me...

**Comment by Playsky** [comment]
hum.. Not really..

- 1 First Thank you so much to take the time to answer all my questions, Shairport-sync is the best airplay on raspberry I know !

- 2 I already saw  this url:https://www.sugrsugr.com/index.php/airplay-prev-next/ And again I don't understand how it works.. ( I'm good in Python but so bad with linux.. ) So I read the tutorial but there is a lot of things I don't understand ( I'm French :) ), so I have to install what, because the tutorial said there is no configuration to do with Ubuntu ? And there is a "patch" at the end of the page which contain 3 files, one is main.c, so how I install this ? where ? I tried to "gcc main.c" but I have a lot of error with the includes so I  have to install things ? Because I already have avahi-utils and other avahi libs ( I think.. ). How can Ido that ? And I found the active remote, user agent and the DACP-ID with the command tcpflow, but there is no informations about the IP and the port which I have to send a request ..

So if you can't answer this last question it's not a problem because you already answered my question in the title, if you can explain me THANKS, and if you so nice :)) Explain me all I have to do because I'm a noob with Linux !

Thanks !

**Comment by mikebrady** [comment]
Thanks for the comments and the kind words. However, the problem that was the cause of the issue has been solved.

You have now moved on to another problem which is not connected with the issue, so it really should be a new issue, and this issue should be closed.

The new issue, unfortunately, is not an issue with Shairport Sync. The link you give makes reference to another version of `shairport`. From the content of your message, and as you have written, you need to learn a lot more about Linux and about programming, and I suggest you look elsewhere for that kind of instruction – this is not the right forum for that.
